homeassistant/components/dsmr/sensor.py

```diff
@@ -381,6 +391,103 @@ class DSMRSensorEntityDescription(
 )
 
 
+def create_mbus_entity(
+    mbus: int, mtype: int, telegram: dict[str, DSMRObject]
+) -> DSMRSensorEntityDescription:
+    """Create a new MBUS Entity."""
+    if mbus == 1:
+        if mtype == 3:
+            obis_reference = obis_references.BELGIUM_MBUS1_METER_READING2
+        elif mtype == 7:
+            obis_reference = obis_references.BELGIUM_MBUS1_METER_READING1
+    elif mbus == 2:
+        if mtype == 3:
+            obis_reference = obis_references.BELGIUM_MBUS2_METER_READING2
+        elif mtype == 7:
+            obis_reference = obis_references.BELGIUM_MBUS2_METER_READING1
+    elif mbus == 3:
+        if mtype == 3:
+            obis_reference = obis_references.BELGIUM_MBUS3_METER_READING2
+        elif mtype == 7:
+            obis_reference = obis_references.BELGIUM_MBUS3_METER_READING1
+    elif mbus == 4:
+        if mtype == 3:
+            obis_reference = obis_references.BELGIUM_MBUS4_METER_READING2
+        elif mtype == 7:
+            obis_reference = obis_references.BELGIUM_MBUS4_METER_READING1
+
+    if mtype == 3:
+        sensor = DSMRSensorEntityDescription(
+            key=f"mbus{mbus}_gas_reading",
+            name=f"Gas consumption mbus{mbus}",
+            obis_reference=obis_reference,
+            is_gas=True,
+            force_update=True,
+            device_class=SensorDeviceClass.GAS,
+            state_class=SensorStateClass.TOTAL_INCREASING,
+        )
+    elif mtype == 7:
+        sensor = DSMRSensorEntityDescription(
```

Comment:
Added a check to if obis_reference is None cause otherwise:
homeassistant/components/dsmr/sensor.py:418: error: Argument "obis_reference" to "DSMRSensorEntityDescription" has incompatible type "Optional[Any]"; expected "str"  [arg-type]